Makefile

```diff
@@ -1,13 +1,21 @@
-.DEFAULT_GOAL := test
+all: test
 
-fmt:
-	./scripts/fmt.sh
 .PHONY: fmt
+fmt: devenv
+	./scripts/fmt.sh
 
-lint:
-	./scripts/lint.sh
 .PHONY: lint
+lint: devenv
+	./scripts/lint.sh
 
+.PHONY: test
 test: fmt lint
 	npm run test
-.PHONY: test
+
+.PHONY: devenv
+devenv: .done/devenv
+
+# indirect through a .done file so that this only runs when deps have changed
+.done/devenv: sbin/install-deps Brewfile jsonnetfile.json package.json
```

Comment:
I'm not following the .done directory/rules here but it seems like an over complication for a simple "install dependencies" step.